File: js/BeginnersSelection/01_Product.js
// https://atcoder.jp/contests/abs/tasks/abc086_a

Object.prototype.let = function(f) {
  return f(this.valueOf());
}

require('fs')
  .readFileSync('/dev/stdin', 'utf8')
  .split(' ')
  .map(x => x & 1)
  .reduce((a, x) => a & x)
  .let(x => x === 1 ? 'Odd' : 'Even')
  .let(console.log);

// https://atcoder.jp/contests/abs/submissions/21477995

File: rb/BeginnersSelection/01_Product.rb
# https://atcoder.jp/contests/abs/tasks/abc086_a

class Object
  def let(f)
    f.call self
  end
end

gets
  .split(" ")
  .map(&:to_i)
  .map { |x| x & 1 }
  .inject { |a, x| a & x }
  .let(->(x) { x == 1 ? 'Odd' : 'Even' })
  .let(method(:puts))

# https://atcoder.jp/contests/abs/submissions/21501400
